app/models/user.rb

```diff
@@ -91,8 +91,12 @@ def not_admin?
     !admin?
   end
 
-  def self.for_email(email)
-    User.find_or_create_by(email_address: email.strip.downcase)
+  def self.for_email(email, client_slug = nil)
```

Comment:
in that case, maybe it makes sense to have a separate method for this? `for_email-and_slug` that calls `for_email` and also does the slug stuff?

 I agree that we are using this method in a lot of places and so it doesn't make sense to change, but I also am worried that we are calling it in a lot of places and doing more than just finding or creating a user by email